src/core/Rule.ml

```diff
@@ -562,6 +562,37 @@ and step = {
 
 and mode_for_step = [ search_mode | taint_mode ] [@@deriving show]
 
+(*****************************************************************************)
+(* Supply chain *)
+(*****************************************************************************)
+
+(* You can only do single layer deep OR *)
+type dependency_formula = dependency_pattern list
+
+(* A pattern to match against versions in a lockfile.
```

Comment:
pattern usually mean a code pattern, so maybe don't use the word pattern here? or maybe
give an example so it's clear what we're talking about.